src/controller.js

```diff
@@ -131,13 +131,15 @@ export default class TreemapController extends DatasetController {
     const meta = this.getMeta();
     const dataset = this.getDataset();
     const dpr = this.chart.currentDevicePixelRatio;
+    const options = this.options;
+    const datasetOptions = this.resolveDatasetElementOptions(mode);
```

Comment:
What this essentially does now, is make the dataset options fall through treemap element options.
The required defaults should just be moved to the controller from element instead.

I hope I did not miss some other side effect.

Reply:
> The required defaults should just be moved to the controller from element instead.

See test case of this PR: https://github.com/kurkle/chartjs-chart-treemap/pull/151/files#diff-9bf86eb940737e9d2f1b9a4743ca173f517e202cd3a1a4cfcb7c7072629fb901

If the user sets options at `elements.treemap` level in chart config, they are ignore, moving defaults in the `controller.defaults`.

Reply:
I have seen that `this.options` (controller/dataset options) has got the fallback to `datasets.treemap` but not to the `elements.treemap`. Instead, the dataset element options (`this.resolveDatasetElementOptions(mode);`) has the fallbacks to `datasets.treemap` and `elements.treemap` but it uses ONLY the keys of `elements.treemap`, ignoring the `datasets.treemap` keys.